utilities/transactions/transaction_lock_mgr.cc

```diff
@@ -491,11 +492,13 @@ bool TransactionLockMgr::IncrementWaiters(
       while (head != -1) {
         assert(wait_txn_map_.Contains(queue_values[head]));
 
+        env->GetCurrentTime(&deadlock_time);
         auto extracted_info = wait_txn_map_.Get(queue_values[head]);
         path.push_back({queue_values[head], extracted_info.m_cf_id,
                         extracted_info.m_waiting_key,
-                        extracted_info.m_exclusive});
+                        extracted_info.m_exclusive, deadlock_time});
```

Comment:
I just noticed that we're repeatedly storing timestamp a bunch of times here.

Instead of extending DeadlockInfo, I think it makes more sense to extend DeadlockPath with a timestamp. That way, the timestamp is associated with the deadlock cycle itself. (instead of every node in the cycle).